pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -16,6 +16,68 @@ final class Event {
       : eventName = DashEvent.analyticsCollectionEnabled,
         eventData = {'status': status};
 
+  /// Event that is emitted when a Dart or Flutter CLI command has been
+  /// executed.
+  ///
+  /// [name] - the name of the command that was executed
+  ///
+  /// [flags] - the set of names for flags and options provided to the command.
+  ///    Does not include values provided by users for options.
+  ///
+  /// [enabledExperiments] - the set of Dart language experiments enabled when
+  ///    running the command.
+  ///
+  /// [exitCode] - the process exit code set as a result of running the command.
+  Event.cliCommandExecuted({
+    required String name,
+    required List<String> flags,
+    required List<String> enabledExperiments,
+    int? exitCode,
+  })  : eventName = DashEvent.cliCommandExecuted,
+        eventData = {
+          'name': name,
+          'flags': flags,
+          'enabledExperiments': enabledExperiments,
+          if (exitCode != null) 'exitCode': exitCode,
+        };
+
+  /// Event that captures timing data associated with some operation.
+  ///
+  /// [operation] - the operation performed that was timed for this event.
+  ///
+  /// [timeMs] - the time the operation took to complete, in milliseconds.
+  Event.timing({
+    required String operation,
+    required int timeMs,
+  })  : eventName = DashEvent.timing,
+        eventData = {
+          'operation': operation,
+          'timeMs': timeMs,
+        };
```

Comment:
I like the idea of having a general timing event but give me a couple of days to check against other data sources if we want to include additional metadata for this

Reply:
I've removed the `timing` event for now to unblock the migration work (we can always add this back into the `pub get` logic later if we think it's useful).